node/src/main/scala/coop/rchain/node/api/WebApi.scala

```diff
@@ -193,19 +201,29 @@ object WebApi {
       usePreStateHash: Boolean
   )
 
-  final case class DataRequest(
+  final case class DataAtNameRequest(
       // For simplicity only one Unforgeable name is allowed
       // instead of the whole RhoExpr (proto Par)
       name: RhoUnforg,
       // Number of blocks in the past to search for data
       depth: Int
   )
 
-  final case class DataResponse(
+  final case class DataAtParRequest(
+      blockHash: String,
+      name: RhoUnforg,
+      usePreStateHash: Boolean
+  )
+
+  final case class DataAtNameResponse(
       exprs: List[RhoExprWithBlock],
       length: Int
   )
 
+  final case class DataAtParResponse(
+      pars: Seq[Par]
+  )
```

Comment:
I am ok to return without `BlockInfo` here.

Reply:
If user didn't specify specific block he will not know from which state is the returned data. But I'd agree that in the future we can reduce amount data about block.

Reply:
I introduced this case class for response as duplicate of `ExploratoryDeployResponse`
https://github.com/rchain/rchain/blob/02edbaa462e7ee0f1c1dda1200e95163be4497d6/node/src/main/scala/coop/rchain/node/api/WebApi.scala#L245-L248

I couldn't find a way, how to add an alias for `ExploratoryDeployResponse`.